File: leetcode-js/289-game-of-life.js
/**
 * @param {number[][]} board
 * @return {void} Do not return anything, modify board in-place instead.
 */
var gameOfLife = function(board) {
    for (let y = 0; y < board.length; y += 1) {
        for (let x = 0; x < board[0].length; x += 1) {
            let count = -board[y][x];
            for (let y1 = Math.max(0, y-1); y1 < Math.min(board.length, y+2); y1 += 1) {
                for (let x1 = Math.max(0, x-1); x1 < Math.min(board[0].length, x+2); x1 += 1) {
                    count += board[y1][x1] & 0b1;
                }
            }
            if ((count | board[y][x]) === 0b11) {
                board[y][x] |= 0b10;
            }
        }
    }

    for (let y = 0; y < board.length; y += 1) {
        for (let x = 0; x < board[0].length; x += 1) {
            board[y][x] >>= 1;
        }
    }
};

File: leetcode-java/289-game-of-life.java
public class Solution {
    public void gameOfLife(int[][] board) {
        int m = board.length;
        int n = board[0].length;

        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                int count = -board[i][j];
                int iLower = Math.max(0, i - 1);
                int iUpper = Math.min(m - 1, i + 1);
                int jLower = Math.max(0, j - 1);
                int jUpper = Math.min(n - 1, j + 1);
                for (int ii = iLower; ii <= iUpper; ii++) {
                    for (int jj = jLower; jj <= jUpper; jj++) {
                        count += board[ii][jj] & 0b1;
                    }
                }
                if ((count | board[i][j]) == 0b11) {
                    board[i][j] |= 0b10;
                }
            }
        }

        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                board[i][j] >>= 1;
            }
        }
    }
}
